### PR TITLE
Meteor module

### PR DESCRIPTION
**Pull request by madushan1000**

### CODE REVIEW

File: lib/modules/meteor/index.js [hunk 1]
```diff
@@ -56,18 +60,117 @@ export function setup(api) {
...
+  const config = api.getConfig().meteor;
+  if (!config) {
+    console.error('error: no configs found for meteor');
+    process.exit(1);
+  }
+
+  var buildLocation = path.resolve('/tmp', uuid.v4());
+  var bundlePath = path.resolve(buildLocation, 'bundle.tar.gz');
+
+  const promise = buildApp(config.path, buildLocation, config.buildOptions || {});
```
Comment (arunoda) [inline lib/modules/meteor/index.js]: Add then directly to the `buildApp(...)` rather creating a new variable.

**Comment by afaur** [comment]
@madushan1000 
Hey there,

I was taking a look at your work to see if I could help in some way.  
I would first like to say that I think this PR is getting kind of big.  

To make it easier to review I stuck a rebased version here temporarily 
(https://github.com/afaur/meteor-up/commit/a3c4f4943fc6d2f3697372efb9e399c4ced34d12).  

While looking at your test errors it looks like on the error for test: `module - meteor setup should setup enviranment on "meteor" vm:` is at step (lib/modules/meteor/__tests__/index.js:21:191), however, since this project is being converted to ES5 behind the scenes I would imagine it might make it difficult to find out what `real` line of code that is.  

I was curious if you are having any trouble with that, or if there is anything I could do to help with this PR.

**Comment by arunoda** [comment]
Hey @afaur it's great you are looking at this. Actually, we were preparing our test setup. That's why this is taking a lot of time and so on. We wanted to test all the mup functionality inside a real server. 

We are still working on that. I assume that's why you tests failed.
I think @madushan1000 can build the test setup by end of next week. With that, we can add more features and port `mupx` faster with reliable manner.

I suggest you to wait a bit until that.

**Comment by afaur** [comment]
@arunoda Understood, looks like your setting up a docker container with keys and ssh'ing into it to test the code which is great.  It always is a lot harder to test deployment type functionalities.

As far as the tests failure I was actually referring to the CircleCi error report from @madushan1000 's recent pushes here:
screen shot 2015-12-18 at 12 23 33 am

By the looks of it I had thought it might prove harder to find out exactly what line of code in his test https://github.com/afaur/meteor-up/commit/a3c4f4943fc6d2f3697372efb9e399c4ced34d12#diff-08e589ca7fff290c72ee44d3f7edff4eR28 would be the offending problem.

All the best, if I can be of any help please let me know.

**Comment by arunoda** [comment]
Hey @madushan1000 I ran tests. Most of the times, they do run. But I found two errors often.

* Once is a AUFS error
* Other one is it's failing tests randomly

See sample logs: https://gist.github.com/arunoda/4b1498d60af5e90b5abc

**Comment by madushan1000** [comment]
I fixed some test script bugs. Now it should work

**Comment by arunoda** [comment]
Hey @afaur now we've an automated testing setup. 
Then we can accept PRs easily.
See more: https://github.com/kadirahq/meteor-up/blob/master/CONTRIBUTING.md

**Comment by arunoda** [comment]
Let me know, what would you like to work on.
Check here: https://waffle.io/kadirahq/meteor-up

**Comment by afaur** [comment]
@arunoda Great, i'll take a look and try to find something.

**Comment by arunoda** [comment]
Nice.

**Pull request merged**